selection_tools/src/Event.cpp

```diff
@@ -230,26 +230,63 @@ vector<tuple<Particle*, Particle*>> Event::get_muon_pair()
 
 tuple<Particle*, Particle*> Event::get_smallest_deltaR_same_sign_muon_non_pair()
 {
-  // TODO: implement that...
   
   tuple<Particle*, Particle*> muons;
   
-  for(auto particle : particles){
-    //...
+  int muon1_index = -1;
+  int muon2_index= -1;
+  double deltaR_max = 0.0;
+  for(int i=0; i<particles.size(); i++){
+    if(!particles[i]->is_good_non_top_muon(particles)) continue;
+    for(int j=0; j<particles.size(); j++)
```

Comment:
I think that the second loop should go from j==(i+1), such that we don't double count.